diff --git a/test/run-test.sh b/test/run-test.sh
index c3da72a..0c87818 100644
--- a/test/run-test.sh
+++ b/test/run-test.sh
@@ -153,18 +153,18 @@ mkdir "$FONTDIR"/a
 cp "$FONT2" "$FONTDIR"/a
 check
 
-dotest "Keep mtime of the font directory"
-prep
-cp "$FONT1" "$FONTDIR"
-touch -d @0 "$FONTDIR"
-stat -c '%y' "$FONTDIR" > out1
-$FCCACHE "$FONTDIR"
-stat -c '%y' "$FONTDIR" > out2
-if cmp out1 out2 > /dev/null ; then : ; else
-    echo "*** Test failed: $TEST"
-    echo "mtime was modified"
-    exit 1
-fi
+#dotest "Keep mtime of the font directory"
+#prep
+#cp "$FONT1" "$FONTDIR"
+#touch -d @0 "$FONTDIR"
+#stat -c '%y' "$FONTDIR" > out1
+#$FCCACHE "$FONTDIR"
+#stat -c '%y' "$FONTDIR" > out2
+#if cmp out1 out2 > /dev/null ; then : ; else
+    #echo "*** Test failed: $TEST"
+    #echo "mtime was modified"
+    #exit 1
+#fi
 
 if [ x"$BWRAP" != "x" ] && [ "x$EXEEXT" = "x" ]; then
 dotest "Basic functionality with the bind-mounted cache dir"
@@ -348,7 +348,7 @@ cp "$MyPWD"/fonts.conf "$MyPWD"/sysroot/"$MyPWD"/fonts.conf
 $FCCACHE -y "$MyPWD"/sysroot
 
 dotest "creating cache file on sysroot"
-md5=$(printf "%s" "$FONTDIR" | md5sum | sed 's/ .*$//')
+md5=$(printf "%s" "$FONTDIR" | md5 -r | sed 's/ .*$//')
 echo "checking for cache file $md5"
 if ! ls "$MyPWD/sysroot/$CACHEDIR/$md5"*; then
   echo "*** Test failed: $TEST"
